lib/api/completeMultipartUpload.js

```diff
@@ -524,35 +524,73 @@ function completeMultipartUpload(authInfo, request, log, callback) {
         function deletePartsMetadata(mpuBucket, keysToDelete, aggregateETag,
             extraPartLocations, destinationBucket, generatedVersionId, droppedMPUSize, next) {
             services.batchDeleteObjectMetadata(mpuBucket.getName(),
-                keysToDelete, log, err => next(err, extraPartLocations,
-                    destinationBucket, aggregateETag, generatedVersionId, droppedMPUSize));
+                keysToDelete, log, err => {
+                    if (err) {
+                        log.error('error deleting MPU metadata during completeMPU', {
+                            method: 'completeMultipartUpload',
+                            mpuBucket: mpuBucket.getName(),
+                            keysToDeleteCount: keysToDelete.length,
+                            error: err,
+                        });
+
+                        // Handle specific error cases according to retry strategy
+                        if (err.is?.DeleteConflict) {
+                            // DeleteConflict should trigger automatic retry
+                            // Convert to InternalError to make it retryable
+                            return next(errors.InternalError, extraPartLocations,
+                                destinationBucket, aggregateETag, generatedVersionId, droppedMPUSize);
+                        }
+
+                        // For NoSuchKey and other errors, return them as-is
+                        // NoSuchKey is non-retryable, InternalError and others are retryable
+                        return next(err, extraPartLocations,
+                            destinationBucket, aggregateETag, generatedVersionId, droppedMPUSize);
```

Comment:
```suggestion
                        if (err.is?.DeleteConflict) {
                            // DeleteConflict should trigger automatic retry. Convert to InternalError to make it retryable
                            err = errors.InternalError
                        }

                        return next(err, extraPartLocations,
                            destinationBucket, aggregateETag, generatedVersionId, droppedMPUSize);
```

An another way is to name variable : `const retryableErrors = errors.InternalError; err = retryableErrors`.